src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -540,11 +541,17 @@ protected override async Task<bool> AcceptCommand(MessageId id, JObject parms, C
                             SendResponse(id, Result.Err("ApplyUpdate failed."), token);
                         return true;
                     }
-                case "DotnetDebugger.addSymbolServerUrl":
+                case "DotnetDebugger.setSymbolOptions":
                     {
-                        string url = args["url"]?.Value<string>();
-                        if (!string.IsNullOrEmpty(url) && !urlSymbolServerList.Contains(url))
-                            urlSymbolServerList.Add(url);
+                        SendResponse(id, Result.OkFromObject(new { }), token);
+                        CachePathSymbolServer = args["symbolOptions"]?["cachePath"]?.Value<string>();
+                        var urls = args["symbolOptions"]?["searchPaths"]?.Value<JArray>();
+                        if (urls == null)
+                            return true;
+                        UrlSymbolServerList.Clear();
+                        UrlSymbolServerList.AddRange(urls.Values<string>());
+                        if (!JustMyCode)
+                            return await LoadSymbolsFromSymbolServer(id, context, token);
```

Comment:
What happens if the user had `!JustMyCode`, and we loaded symbols from the symbol server. Then the user sets `JustMyCode=true`, and we still have the symbols from the symbol server loaded. How does that affect the UX?

Reply:
This method ShouldSkipMethod together with the new attribute HasNonUser code and the JustMyCode setting will know that even having the symbols, as it's not a user code and the justmycode is enabled, the method should be skipped while stepping.